programs/manifest/src/program/processor/global_clean.rs

```diff
@@ -0,0 +1,150 @@
+use std::cell::RefMut;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use hypertree::{get_helper, trace, DataIndex, RBNode};
+use solana_program::{account_info::AccountInfo, entrypoint::ProgramResult, pubkey::Pubkey};
+
+use crate::{
+    program::{batch_update::MarketDataTreeNodeType, get_mut_dynamic_account, ManifestError},
+    quantities::{GlobalAtoms, WrapperU64},
+    require,
+    state::{
+        utils::get_now_slot, GlobalRefMut, MarketRefMut, OrderType, RestingOrder, MARKET_BLOCK_SIZE,
+    },
+    validation::loaders::{GlobalCleanContext, GlobalTradeAccounts},
+};
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct GlobalCleanParams {
+    pub order_index: DataIndex,
+}
+
+impl GlobalCleanParams {
+    pub fn new(order_index: DataIndex) -> Self {
+        GlobalCleanParams { order_index }
+    }
+}
+
+pub(crate) fn process_global_clean(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    data: &[u8],
+) -> ProgramResult {
+    trace!("process_global_clean accs={accounts:?}");
+    let global_clean_context: GlobalCleanContext = GlobalCleanContext::load(accounts)?;
+
+    let GlobalCleanContext {
+        payer,
+        market,
+        global,
+        system_program,
+    } = global_clean_context;
+
+    let global_trade_accounts: GlobalTradeAccounts = GlobalTradeAccounts {
+        mint_opt: None,
+        global: global.clone(),
+        global_vault_opt: None,
+        market_vault_opt: None,
+        token_program_opt: None,
+        system_program: Some(system_program),
+        trader: payer,
```

Comment:
is this line what enables the indirect transfer from the deposit an user made for his order that is not being cleaned to the person invoking global clean?

Reply:
Kinda. There is only one account info that gives or receives gas, the signer. We don't even have the other persons account info. Can document and rename to something else if you think of something better

Reply:
This passes in the account info and then place and cancel credit or debit it with gas payments

Reply:
yeah the main confusion here comes from the re-use of the trader field, which intuitively would be the person that placed the order with the payer identity which really could be someone who never trades

Reply:
in general it's not ideal if a method has 10+ arguments and a change in one of them (especially hidden in a struct) can cause critical bugs

Reply:
i would favor a more explicit code style

Reply:
or do you want to throw away the "give me a ref to everything I could need for global" struct altogether

Reply:
gas_receiver_opt & gas_payer_opt definitely an improvement. i thought to first split the method and then have another look at the struct how it's used post-merge to see what's used where, so we can make a decision how to break it up

Reply:
which method? this everything struct is passed through for both place and cancel

Reply:
remove_from_global

Reply:
are you still planning to split out gas_receiver into it's own account?